### PR TITLE
One Body Density Matrices batched driver integration

### PR DESCRIPTION
**Pull request by PDoakORNL**
## Proposed changes
Integration of One Body Density Matrices with unified driver.
Fix of SpinDensityNew test so corner = 0,0,0 which is what was used for reference.

Fix so that the found python will actually be used to check for installed python packages.
in the context of execute_process it is quite likely that the `#!/usr/bin/env python3` in the test_import.py script will not use the intended and found python.  This causes issues with conda and spack installed packages.
Obviously find_package(Python3) must find your intended python.

Factor our handling of center and corner parameters in estimator input.

## What type(s) of changes does this code introduce?

- New feature
- Refactoring (no functional changes, no api changes)
- Build related changes
- Testing changes (e.g. new unit/integration/performance tests)
- Documentation changes

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?
Leconte

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes This PR is up to date with current the current state of 'develop'
- Yes Code added or changed in the PR has been clang-formatted
- Yes This PR adds tests to cover any new code, or to catch a bug that is being fixed
- Yes Documentation has been added (if appropriate)

### CODE REVIEW

**Comment by PDoakORNL** [comment]
Some rebase damage.  Will reopen when fixed.

**Comment by PDoakORNL** [comment]
Actually found uniform_grid integrator path broken, now that is fixed as well.

**Comment by prckent** [comment]
Very interested to see the python fix included. Possibly all the deterministic errors from nitrogen & sulfur are due to this.

**Comment by PDoakORNL** [comment]
I think my fix only deals with detection of python packages.

File: tests/solids/diamondC_1x1x1_pp/CMakeLists.txt [hunk 1]
```diff
@@ -463,6 +463,29 @@ if(add_estimator_tests)
   #   check_stats.py -s 0 -q spindensity -e 20 -c 8 -p qmc_spindens_short -r qmc-ref/qmc_spindens_short.s000.stat_ref_spindensity.dat
   #   )
 
+  # # Unlikey check_states.py will be able to digest this
```
Comment (ye-luo) [inline tests/solids/diamondC_1x1x1_pp/CMakeLists.txt:466]: Unlikey is a typo and this message doesn't help me understand what is unique about this test.

**Review by ye-luo — CHANGES_REQUESTED** [review]
I don't feel right that MCPopulation is used to pass WaveFunctionFactory but that is more of refactoring MCPopulation handling. Please fix the minor issues and then we can merge this PR.

**Comment by PDoakORNL** [comment]
I don't feel great about the wf_factory in MCPopulation either but estimators need to be able to get SPOSets.

**Comment by ye-luo** [comment]
Start testing in-house

**Review by ye-luo — APPROVED** [review]

**Pull request merged**